### PR TITLE
Fix videoclip.mask return None when use apply_to_mask decorator

### PR DESCRIPTION
**Pull request by biandh**
before

```python
@decorator.decorator
def apply_to_mask(f, clip, *a, **k):
    """ This decorator will apply the same function f to the mask of
        the clip created with f """

    newclip = f(clip, *a, **k)
    if getattr(newclip, "mask", None):
        newclip.mask = f(newclip.mask, *a, **k)
    return newclip
```

after

```python
@decorator.decorator
def apply_to_mask(f, clip, *a, **k):
    """ This decorator will apply the same function f to the mask of
        the clip created with f """

    newclip = f(clip, *a, **k)
    if getattr(clip, "mask", None):  # newclip replace with clip
        newclip.mask = f(clip.mask, *a, **k)
    return newclip
```

Use newclip as a condition in some cases this can cause exceptions，for example:

```python
    test_file = "resource/sintel_5s.mp4"  # duration=5
    test_clip = VideoFileClip(test_file, has_mask=True)
    test_clip = test_clip.loop(duration=test_clip.duration * 2)

    video_file = "resource/sintel_with_15_chapters.mp4"
    video_clip = VideoFileClip(video_file).subclip(0, test_clip.duration)

    final_clip = CompositeVideoClip([video_clip, test_clip])
    final_clip.write_videofile("resource/mask_test.mp4", audio=False)
```

This  failed with the following error:

```
t:  49%|████▉     | 118/242 [00:03<00:03, 36.95it/s, now=None]Traceback (most recent call last):
  File "/Users/biandh/github_workspace/moviepy/tests/test_VideoFileClip.py", line 72, in <module>
    test_video_mask()
  File "/Users/biandh/github_workspace/moviepy/tests/test_VideoFileClip.py", line 67, in test_video_mask
    final_clip.write_videofile('resource/xxx.mp4', audio=False)
  File "<decorator-gen-55>", line 2, in write_videofile
  File "/Users/biandh/github_workspace/moviepy/moviepy/decorators.py", line 55, in requires_duration
    return f(clip, *a, **k)
  File "<decorator-gen-54>", line 2, in write_videofile
  File "/Users/biandh/github_workspace/moviepy/moviepy/decorators.py", line 128, in use_clip_fps_by_default
    return f(clip, *new_a, **new_kw)
  File "<decorator-gen-53>", line 2, in write_videofile
  File "/Users/biandh/github_workspace/moviepy/moviepy/decorators.py", line 23, in convert_masks_to_RGB
    return f(clip, *a, **k)
  File "/Users/biandh/github_workspace/moviepy/moviepy/video/VideoClip.py", line 341, in write_videofile
    logger=logger,
  File "/Users/biandh/github_workspace/moviepy/moviepy/video/io/ffmpeg_writer.py", line 254, in ffmpeg_write_video
    logger=logger, with_times=True, fps=fps, dtype="uint8"
  File "/Users/biandh/github_workspace/moviepy/moviepy/Clip.py", line 470, in iter_frames
    frame = self.get_frame(t)
  File "<decorator-gen-11>", line 2, in get_frame
  File "/Users/biandh/github_workspace/moviepy/moviepy/decorators.py", line 87, in wrapper
    return f(*new_a, **new_kw)
  File "/Users/biandh/github_workspace/moviepy/moviepy/Clip.py", line 97, in get_frame
    return self.make_frame(t)
  File "/Users/biandh/github_workspace/moviepy/moviepy/video/compositing/CompositeVideoClip.py", line 113, in make_frame
    f = c.blit_on(f, t)
  File "/Users/biandh/github_workspace/moviepy/moviepy/video/VideoClip.py", line 598, in blit_on
    mask = self.mask.get_frame(ct) if self.mask else None
  File "<decorator-gen-11>", line 2, in get_frame
  File "/Users/biandh/github_workspace/moviepy/moviepy/decorators.py", line 87, in wrapper
    return f(*new_a, **new_kw)
  File "/Users/biandh/github_workspace/moviepy/moviepy/Clip.py", line 97, in get_frame
    return self.make_frame(t)
  File "/Users/biandh/github_workspace/moviepy/moviepy/Clip.py", line 140, in <lambda>
    newclip = self.set_make_frame(lambda t: fun(self.get_frame, t))
  File "/Users/biandh/github_workspace/moviepy/moviepy/Clip.py", line 192, in <lambda>
    lambda gf, t: gf(t_func(t)), apply_to, keep_duration=keep_duration
  File "<decorator-gen-11>", line 2, in get_frame
  File "/Users/biandh/github_workspace/moviepy/moviepy/decorators.py", line 87, in wrapper
    return f(*new_a, **new_kw)
  File "/Users/biandh/github_workspace/moviepy/moviepy/Clip.py", line 97, in get_frame
    return self.make_frame(t)
  File "/Users/biandh/github_workspace/moviepy/moviepy/video/io/VideoFileClip.py", line 117, in <lambda>
    mask_mf = lambda t: self.reader.get_frame(t)[:, :, 3] / 255.0
  File "/Users/biandh/github_workspace/moviepy/moviepy/video/io/ffmpeg_reader.py", line 204, in get_frame
    result = self.read_frame()
  File "/Users/biandh/github_workspace/moviepy/moviepy/video/io/ffmpeg_reader.py", line 158, in read_frame
    % (self.filename)
OSError: MoviePy error: failed to read the first frame of video file resource/sintel_5s.mp4. That might mean that the file is corrupted. That may also mean that you are using a deprecated version of FFMPEG. On Ubuntu/Debian for instance the version in the repos is deprecated. Please update to a recent version from the website.
```

Through debugging, I found out that it is caused by `newclip.mask = None`, so I think it needs to be fixed here.

### CODE REVIEW

File: tests/test_VideoFileClip.py [hunk 1]
```diff
@@ -45,5 +46,28 @@ def test_ffmpeg_resizing():
         video.close()
 
 
+def test_video_mask():
+    """Test clip mask attributes operate."""
+    test_file = "resource/sintel_with_15_chapters.mp4"
+
+    return
```
Comment (mgaitan) [inline tests/test_VideoFileClip.py:53]: this makes the test does nothing, right?

Comment (biandh) [reply tests/test_VideoFileClip.py:53]: I didn't do this intentionally, but during the test, there will be an error that the resource/sintel_with_15_chapters.mp4 file cannot be found, which is a bit confusing. I don't know why this file cannot be found. Do you know what caused this?

Comment (biandh) [reply tests/test_VideoFileClip.py]: I seem to know the solution to the problem that occurred during the test! I have rewritten the test script.

**Comment by OsaAjani** [comment]
Thank you for your contributions and for reporting issues in this repository. With the release of v2, which introduces significant changes to the codebase and API, we’ve reviewed the backlog of open PRs and issues. Due to the length of the backlog and the likelihood that many of these are either fixed or no longer applicable, we’ve made the decision to close all previous PRs and issues.

If you believe that any of these are still relevant to the current version or if you'd like to reopen a related discussion, please feel free to create a new issue or pull request, referencing the old one.

Thank you for your understanding and continued support!

**Pull request closed without merging**